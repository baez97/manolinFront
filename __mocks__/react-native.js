function AsyncStorageClass() {
    this.setItem = jest.fn((key, value) => {
        return new Promise((resolve, reject) => {
            resolve();
        })
    });

    this.getItem = jest.fn((key, value) => {
        return new Promise((resolve, reject) => {
            resolve("THIS_IS_THE_SAVED_TOKEN");
        })
    });

    this.removeItem = jest.fn((key) => {
        return  new Promise((resolve, reject) => {
            resolve();
        })
    });

    this.removeItem.mockReturnValueOnce(new Promise((res,rej) => {
        rej({message: "THIS IS AN ERROR"});
    }))
}

module.exports.AsyncStorage = new AsyncStorageClass();