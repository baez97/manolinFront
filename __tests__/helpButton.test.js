import React from 'react';
import { Linking }Â from 'react-native';
import renderer from 'react-test-renderer';
import HelpButton from '../components/HelpButton';

jest.unmock("react-native")
jest.unmock('react')
jest.unmock("react-test-renderer")
jest.unmock('expo');

describe("HelpButton", () => {
    describe("Methods", () => {
        const hB = new HelpButton();

        it("Calls to Linking when it is pressed", () => {
            Linking.openURL = jest.fn();
            hB.handlePress();

            expect(Linking.openURL).toBeCalled();
        })
    });

    describe("View", () => {
        it("Matches the Snapshot", () => {
            const view = <HelpButton text="TEXT" />
            const rendered = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        })
    })
})