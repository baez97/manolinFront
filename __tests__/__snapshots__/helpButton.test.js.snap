// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpButton View Matches the Snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4294967295,
        4294111986,
      ]
    }
    endPoint={
      Array [
        0.65,
        1,
      ]
    }
    startPoint={
      Array [
        0.55,
        0,
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 53.25,
        "elevation": 3,
        "height": 84.89999999999999,
        "justifyContent": "center",
        "paddingLeft": 52.5,
        "paddingRight": 52.5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#23538a",
          "fontFamily": "montserrat-extra-bold",
          "fontSize": 25.50000071525972,
        }
      }
    >
      TEXT
    </Text>
  </ExponentLinearGradient>
</View>
`;
